### PR TITLE
New metadata service

### PR DESCRIPTION
**Pull request by namelessssssssssss**
## Purpose of the Change
This PR introduces a new version of `MetadataService` (V2) based on Protobuf, which provides better compatibility with dubbo-go.

See https://github.com/apache/dubbo/issues/13584

## Brief Changelog
To use `MetadataServiceV2`, the application must meet the following requirements:
* The application has imported Protobuf dependencies.
* `metadataServiceProtocol` is configured as 'tri', or at least one service is using the triple protocol.

Currently, `MetadataServiceV2` will automatically export along with `MetadataService` (V1, on the same port) if the above requirements are **satisfied**. If `onlyUseMetadataV2` is set to `true`, only `MetadataServiceV2` will be exported. If the above requirements are **not satisfied**, Dubbo will still only export `MetadataService` (V1).

### CODE REVIEW

**Comment by chickenlj** [comment]
Well done. Generally looks good to me.

**Review by FoghostCn — COMMENTED** [review]
Can we implement the tri protocol in a non-idl way, so that our interface definition can be reused, and we only need to set the protocol to tri when using it. I don't know if this is feasible.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/ServiceInstanceMetadataUtils.java [hunk 1]
```diff
@@ -96,6 +96,8 @@ public class ServiceInstanceMetadataUtils {
      */
     public static final String METADATA_STORAGE_TYPE_PROPERTY_NAME = "dubbo.metadata.storage-type";
 
+    public static final String METADATA_SERVICE_VERSION_NAME = "dubbo.metadata.service-version";
```
Comment (chickenlj) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/ServiceInstanceMetadataUtils.java]: Use as short a name as possible, for example, `meta-v`.

**Review by chickenlj — APPROVED** [review]
LGTM.

File: dubbo-metadata/dubbo-metadata-api/src/main/proto/metadata_service_v2.proto [hunk 2]
```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
```
Comment (AlbumenJ) [inline dubbo-metadata/dubbo-metadata-api/src/main/proto/metadata_service_v2.proto:18]: License header is required

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-metadata/dubbo-metadata-api/src/main/proto/metadata_service_v2.proto (hunk 2 above)
Comment (namelessssssssssss) [reply dubbo-metadata/dubbo-metadata-api/src/main/proto/metadata_service_v2.proto:18]: OK. License are added now

**Pull request merged**